src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1167,6 +1167,16 @@ private bool IsValidFixedVariableInitializer(TypeSymbol declType, SourceLocalSym
                 Error(diagnostics, ErrorCode.ERR_ManagedAddr, initializerSyntax, elementType);
                 hasErrors = true;
             }
+            else if (elementType.GetArity() != 0)
+            {
+                var supported = MessageID.IDS_FeatureUnmanagedGenericStructs.RequiredVersion() >= Compilation.LanguageVersion;
```

Comment:
simplify

Reply:
Looks like that is because the check here is always redundant. Not sure which is unnecessary but suspect this one is.

Reply:
I think the check here is necessary due to the fact that you can use an array as a `fixed` initializer.

If you remove this check, you won't get an error for using an expression of type `MyStruct<int>[]` for the `fixed` initializer. If you remove the `hasErrors = true` you will get a redundant ERR_FeatureNotAvailable for `fixed` statements that use the `&` operator in the initializer.

Maybe it would be helpful (possibly more clear?) to only do this check for fixed initializers when the initializer is *not* an AddressOf expression?

Reply:
I'm now using the Binder.CheckFeatureAvailability API which lets you know if the feature is supported or not